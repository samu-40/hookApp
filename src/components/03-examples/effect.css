body{
    padding: 20px;
}

.btn-outline-success{
    margin-left: 10px;
}