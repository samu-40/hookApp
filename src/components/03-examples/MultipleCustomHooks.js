import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import './effect.css';


export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);

    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    const {author, quote} = !!data && data[0];
    //                      |________________|
    //                              |
    // Si data tiene algo entonces extrae el elemento cero de esta.

    return (
        <div>
            <h3>Breaking Bads: Quotes</h3>
            <hr />

            {
                loading ?
                    (
                        <div className='alert alert-info text-center'>
                            Loading...
                        </div>
                    )
                    :
                    (
                        <figure className='text-end'>
                            <blockquote className="blockquote">
                                <p>{quote}</p>
                            </blockquote>
                            <figcaption className="blockquote-footer">
                                {author}
                            </figcaption>
                        </figure>
                    )
            }

            <button 
                className='btn btn-secondary'
                onClick={increment}
            >
                Next quote
            </button>
        </div>
    );

};
