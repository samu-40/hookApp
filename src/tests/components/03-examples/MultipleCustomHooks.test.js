import {shallow} from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');

describe(',Pruebas en MultipleCustomHooks.js', () => {
    
    test('Debe mostrarse correctamente', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null,
        });

        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrarse la info', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Samuel',
                quote: 'Hola mundo',
            }],
            loading: false,
            error: null,
        });

        const wrapper = shallow(<MultipleCustomHooks/>);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.blockquote').text().trim()).toBe('Hola mundo');
        expect(wrapper.find('.blockquote-footer').text().trim()).toBe('Samuel');
    });
    

});
