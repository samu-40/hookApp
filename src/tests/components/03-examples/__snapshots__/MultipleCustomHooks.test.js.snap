// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`,Pruebas en MultipleCustomHooks.js Debe mostrarse correctamente 1`] = `
<div>
  <h3>
    Breaking Bads: Quotes
  </h3>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
